internal/services/contracts_events_consumer.go

```diff
@@ -602,9 +602,50 @@ func (c *ContractsEventsConsumer) handleAftermarketDeviceAttributeSetEvent(ctx c
 func (c *ContractsEventsConsumer) handlePermissionsSetEvent(ctx context.Context, e *cmodels.ContractEventData) error {
 	logger := c.log.With().Str("EventName", PermissionsSetEvent.String()).Logger()
 
+	// Try to detect which event format by checking the JSON structure
+	var rawArgs map[string]interface{}
+	if err := json.Unmarshal(e.Arguments, &rawArgs); err != nil {
+		return fmt.Errorf("error unmarshaling PermissionsSet raw inputs: %w", err)
+	}
+
+	var templateID []byte
 	var args PermissionsSetData
-	if err := json.Unmarshal(e.Arguments, &args); err != nil {
-		return fmt.Errorf("error unmarshaling PermissionsSet inputs: %w", err)
+
```

Comment:
I would consider extracting the code of  Check if TemplateId and using the new format in a method or function.